ValveKeyValue/ValveKeyValue/Deserialization/KeyValues3/KV3TokenReader.cs

```diff
@@ -0,0 +1,437 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using ValveKeyValue.KeyValues3;
+using Encoding = ValveKeyValue.KeyValues3.Encoding;
+
+namespace ValveKeyValue.Deserialization.KeyValues3
+{
+    class KV3TokenReader : KVTokenReader
+    {
+        const char ObjectStart = '{';
+        const char ObjectEnd = '}';
+        const char BinaryBlobMarker = '#';
+        const char ArrayStart = '[';
+        const char ArrayEnd = ']';
+        const char CommentBegin = '/';
+        const char Assignment = '=';
+        const char Comma = ',';
+
+        public KV3TokenReader(TextReader textReader) : base(textReader)
+        {
+            // Dota 2 binary from 2017 used "+" as a terminate (for flagged values), but then they changed it to "|"
```

Comment:
If Valve was correct about it they would've bumped the text encoding version (guid) when they did this change.🤔